frontend/src/store/transaction.js

```diff
@@ -49,30 +50,84 @@ export default ({
       context.commit('setUser', response.data)
     },
     getRegisteredNodes (context, params) {
-      let page = params.page || context.state.nodePage
+      let page = context.state.nodePage
       if (!page) return
 
-      tfService.getNodes(undefined, params.size, page).then(response => {
+      const getNodes = (page) => {
+        return tfService.getNodes(undefined, params.size, page)
+      }
+
+      getNodes(page).then(response => {
         context.commit('setRegisteredNodes', response)
```

Comment:
why do we commit here ? shouldn't we just commit once we have fetch all the pages?

Reply:
not sure, maybe we can commit in when the promises are resolved

Reply:
Doesn't work well, going to stick with what we have right now.